File: sdks/ruby/lib/novabilling/subscriptions/types/update_subscription_dto.rb
# frozen_string_literal: true

module Novabilling
  module Subscriptions
    module Types
      class UpdateSubscriptionDto < Internal::Types::Model
        field :id, -> { String }, optional: false, nullable: false
        field :metadata, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false
      end
    end
  end
end

File: sdks/php/src/Subscriptions/Requests/UpdateSubscriptionDto.php
<?php

namespace NovaBilling\Subscriptions\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class UpdateSubscriptionDto extends JsonSerializableType
{
    /**
     * @var ?array<string, mixed> $metadata
     */
    #[JsonProperty('metadata'), ArrayType(['string' => 'mixed'])]
    public ?array $metadata;

    /**
     * @param array{
     *   metadata?: ?array<string, mixed>,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->metadata = $values['metadata'] ?? null;
    }
}
